crates/red_knot_python_semantic/src/types/call.rs

```diff
@@ -205,19 +161,94 @@ impl<'db> CallOutcome<'db> {
         match self {
             Self::Callable { binding } => {
                 binding.report_diagnostics(context, node);
-                Ok(binding.return_type())
-            }
-            Self::RevealType {
-                binding,
-                revealed_ty,
-            } => {
-                binding.report_diagnostics(context, node);
-                context.report_diagnostic(
-                    node,
-                    DiagnosticId::RevealedType,
-                    Severity::Info,
-                    format_args!("Revealed type is `{}`", revealed_ty.display(context.db())),
-                );
+
+                if !binding.has_binding_errors() {
+                    if let Type::FunctionLiteral(function_type) = binding.callable_type() {
+                        if let Some(known_function) = function_type.known(context.db()) {
+                            // TODO: Should we skip those diagnostics if there's any binding error?
```

Comment:
Yes, I think it's fine to do that, would be fine to remove this TODO IMO. It's not always clear what the right reveal/assert diagnostic would even be if we had an error binding the call, and the binding error should tell the user what they need to fix in order to get the desired diagnostic.